server/keyshare/keyshareserver/postgresdb.go

```diff
@@ -26,11 +26,15 @@ const emailTokenValidity = 24 // amount of time user's email validation token is
 // var so that tests may change it.
 var backoffStart int64 = 60
 
-func newPostgresDB(connstring string) (DB, error) {
+// newPostgresDB opens a new database connection using the given maximum connection bounds.
+// If maxOpenConns is set to 0, then an unlimited number of connections is allowed.
+func newPostgresDB(connstring string, maxIdleConns int, maxOpenConns int) (DB, error) {
 	db, err := sql.Open("pgx", connstring)
 	if err != nil {
 		return nil, err
 	}
+	db.SetMaxIdleConns(maxIdleConns)
+	db.SetMaxOpenConns(maxOpenConns)
```

Comment:
Open connections will not be closed by the server, because the `ConnMaxLifetime` is unlimited by default. We have to check whether this is a problem. Otherwise, we have to add configuration options for lifetime too.

https://pkg.go.dev/database/sql#DB.SetConnMaxIdleTime

Reply:
Both of these would be good to have. If you like, I can add them myself to this PR.